src/ng-select/items-list.ts

```diff
@@ -310,6 +310,15 @@ export class ItemsList {
         }
     }
 
+    private get _lastMarkedIndex() {
+        const selectedIndex = this._filteredItems.indexOf(this.lastSelectedItem);
+        if (selectedIndex === -1) {
+            this._markedIndex = selectedIndex;
```

Comment:
1. Better convert it to method as it does some search and other operations.

```ts
private _getLastMarkedIndex(): number {}
```
2.  Getter should not modify state as it does here with `this._markedIndex = selectedIndex;`